langchain/llms/base.py

```diff
@@ -255,19 +257,45 @@ async def _agenerate_helper(
         stop: Optional[List[str]],
         run_managers: List[AsyncCallbackManagerForLLMRun],
         new_arg_supported: bool,
+        timeout: Optional[float] = None,
+        abort_signal: Optional[asyncio.Future] = None,
         **kwargs: Any,
     ) -> LLMResult:
         try:
-            output = (
-                await self._agenerate(
+            coroutine = (
+                self._agenerate(
                     prompts,
                     stop=stop,
                     run_manager=run_managers[0] if run_managers else None,
                     **kwargs,
                 )
                 if new_arg_supported
-                else await self._agenerate(prompts, stop=stop)
+                else self._agenerate(prompts, stop=stop)
+            )
+            task = asyncio.create_task(coroutine)
+            tasks_to_await = {
```

Comment:
ooc (out of curiosity) any reason to make this a set vs just a list/tuple?

Reply:
list tuple works too, just wanted to eliminate possibility of duplicate tasks ever (which is unlikely to happen tho even if we use a list)